.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,105 @@
+  CUDA_DOCKER_IMAGE: seracllnl/tpls:cuda-12_08-15-24_21h-49m
+  GCC_DOCKER_IMAGE: seracllnl/tpls:gcc-13_08-15-24_21h-49m
+
+jobs:
+  # Hacky solution to reference env variables outside of `run` steps https://stackoverflow.com/a/74217028
+  set_image_vars:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Do Nothing
+      run: echo
+    outputs:
+      clang_docker_image: ${{ env.CLANG_DOCKER_IMAGE }}
+      cuda_docker_image: ${{ env.CUDA_DOCKER_IMAGE }}
+      gcc_docker_image: ${{ env.GCC_DOCKER_IMAGE }}
+  build_and_test:
+    runs-on: ubuntu-22.04
+    needs:
+    - set_image_vars
+    strategy:
+      matrix:
+        build_type: [ Debug, Release ]
+        config:
+        - host_config: clang@14.0.0.cmake
+          compiler_image: ${{ needs.set_image_vars.outputs.clang_docker_image }}
+          cmake_opts: "-DBUILD_SHARED_LIBS=ON"
+        - host_config: gcc@12.3.0_cuda.cmake
+          compiler_image: ${{ needs.set_image_vars.outputs.cuda_docker_image }}
+          cmake_opts: "-DBUILD_SHARED_LIBS=ON -DENABLE_WARNINGS_AS_ERRORS=OFF"
+          build_src_opts: "--skip-install --skip-tests"
+        - host_config: gcc@13.1.0.cmake
+          compiler_image: ${{ needs.set_image_vars.outputs.gcc_docker_image }}
+          cmake_opts: "-DBUILD_SHARED_LIBS=ON"
+        include:
+        - build_type: Debug
+          config:
+            host_config: clang@14.0.0.cmake
+            compiler_image: ${{ needs.set_image_vars.outputs.clang_docker_image }}
+            cmake_opts: "-DSERAC_ENABLE_CODEVELOP=ON"
+            build_src_opts: "--skip-install"
+    container:
+      image: ${{ matrix.config.compiler_image }}
+      volumes:
+      - /home/serac/serac
+    steps:
+      - name: Checkout Serac
+        uses: actions/checkout@v4
+        with: 
+          submodules: recursive
+      - name: Print Matrix Variables
+        run: |
+          echo "build_src_opts ${{ matrix.config.build_src_opts }}"
+          echo "build_type     ${{ matrix.build_type }}"
+          echo "cmake_opts     ${{ matrix.config.cmake_opts }}"
+          echo "compiler_image ${{ matrix.config.compiler_image }}"
+          echo "host_config    ${{ matrix.config.host_config }}"
+      - name: Build and Test ${{ matrix.config.host_config }} ${{ matrix.build_type }}
+        timeout-minutes: 30
+        run: |
+          ./scripts/llnl/build_src.py ${{ matrix.config.build_src_opts }} \
+            --verbose \
+            --host-config ${{ matrix.config.host_config }} \
+            --extra-cmake-options "${{ matrix.config.cmake_opts }} -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}" \
+            --jobs 4
+      - name: Upload Test Results
+        uses: actions/upload-artifact@v4
+        with:
+          name: Test Results ${{ matrix.config.host_config }} ${{ matrix.build_type }} ${{ matrix.config.cmake_opts }}
+          path: "**/Test.xml"
+          if-no-files-found: ignore # Missing test file is expected on CUDA builds
```

Comment:
What happens if the non-CUDA builds don't generate files?

Reply:
it would be ignored. we might be able to use if statements or revert to using the dummy xml file

Reply:
i would prefer either of those. This makes the base case to allow no tests to be run without reporting.